optuna/integration/sklearn.py

```diff
@@ -0,0 +1,925 @@
+from __future__ import absolute_import
+
+from logging import DEBUG
+from logging import INFO
+from logging import Logger  # NOQA
+from logging import WARNING
+from numbers import Number
+from time import time
+
+import numpy as np
+import pandas as pd  # NOQA
+from scipy.sparse import spmatrix  # NOQA
+
+try:
+    from sklearn.base import BaseEstimator
+    from sklearn.base import clone
+    from sklearn.base import is_classifier
+    from sklearn.metrics import check_scoring
+    from sklearn.model_selection._validation import _index_param_value
+    from sklearn.model_selection import BaseCrossValidator  # NOQA
+    from sklearn.model_selection import check_cv
+    from sklearn.model_selection import cross_validate
+    from sklearn.utils import check_random_state
+    from sklearn.utils.metaestimators import _safe_split
+    from sklearn.utils import safe_indexing as sklearn_safe_indexing
+    from sklearn.utils.validation import _num_samples
+    from sklearn.utils.validation import check_is_fitted
+
+    _available = True
+
+except ImportError as e:
+    BaseEstimator = object
+
+    _import_error = e
+    _available = False
+
+from optuna import distributions
+from optuna import logging
+from optuna import pruners  # NOQA
+from optuna import samplers  # NOQA
+from optuna import storages  # NOQA
+from optuna import structs
+from optuna import study as study_module
+from optuna import trial as trial_module  # NOQA
+from optuna import type_checking
+
+if type_checking.TYPE_CHECKING:
+    from typing import Any  # NOQA
+    from typing import Callable  # NOQA
+    from typing import Dict  # NOQA
+    from typing import List  # NOQA
+    from typing import Mapping  # NOQA
+    from typing import Optional  # NOQA
+    from typing import Union  # NOQA
+
+    OneDimArrayLikeType = Union[List[float], np.ndarray, pd.Series]
+    TwoDimArrayLikeType = \
+        Union[List[List[float]], np.ndarray, pd.DataFrame, spmatrix]
+
+logger = logging.get_logger(__name__)
+
+
+def _check_sklearn_availability():
+    # type: () -> None
+
+    if not _available:
+        raise ImportError(
+            'scikit-learn is not available. Please install scikit-learn to '
+            'use this feature. scikit-learn can be installed by executing '
+            '`$ pip install scikit-learn>=0.20.0`. For further information, '
```

Comment:
Would you mind checking the sklearn version in `sklearn.py`?

FYI: See also an example of version check in the visualization module.
https://github.com/pfnet/optuna/blob/master/optuna/visualization.py#L107-L112